src/librustc_data_structures/sharded.rs

```diff
@@ -137,6 +137,20 @@ impl<K: Eq + Hash + Copy> ShardedHashMap<K, ()> {
     }
 }
 
+pub unsafe trait IntoPointer {
```

Comment:
Documenting the proof obligations would be good.

Reply:
I'll just make it safe instead =P